core/src/main/java/org/arquillian/smart/testing/configuration/Configuration.java

```diff
@@ -0,0 +1,267 @@
+    }
+
+    public RunMode getMode() {
+        return mode;
+    }
+
+    public void setMode(RunMode mode) {
+        this.mode = mode;
+    }
+
+    public String getApplyTo() {
+        return applyTo;
+    }
+
+    public void setApplyTo(String applyTo) {
+        this.applyTo = applyTo;
+    }
+
+    public Boolean isDisable() {
+        return disable != null && disable;
+    }
+
+    public void setDisable(Boolean disable) {
+        this.disable = disable;
+    }
+
+    public Boolean isDebug() {
+        return debug != null && debug;
+    }
+
+    public void setDebug(Boolean debug) {
+        this.debug = debug;
+    }
+
+    public Report getReport() {
+        return report;
+    }
+
+    public void setReport(Report report) {
+        this.report = report;
+    }
+
+    public Scm getScm() {
+        return scm;
+    }
+
+    public void setScm(Scm scm) {
+        this.scm = scm;
+    }
+
+    static Configuration fromSystemProperties() {
+        final Configuration configuration = new Configuration();
+
+        if (System.getProperty(SMART_TESTING) != null) {
+            final String strategies = System.getProperty(SMART_TESTING).toLowerCase();
+
+            if (containsAnyStrategy(strategies)) {
+                configuration.strategies = strategies.split("\\s*,\\s*");
+            }
+        }
+
+        if (System.getProperty(SMART_TESTING_MODE) != null) {
+            configuration.mode = RunMode.valueOf(System.getProperty(SMART_TESTING_MODE).toUpperCase());
+        }
+
+        if (System.getProperty(SMART_TESTING_DISABLE) != null) {
+            configuration.disable = Boolean.valueOf(System.getProperty(SMART_TESTING_DISABLE));
+        }
+
+        if (System.getProperty(SMART_TESTING_DEBUG) != null) {
+            configuration.debug = Boolean.valueOf(System.getProperty(SMART_TESTING_DEBUG));
+        }
+
+        configuration.applyTo = System.getProperty(SMART_TESTING_APPLY_TO);
+
+        configuration.report = Report.fromSystemProperties();
+        configuration.scm = Scm.fromSystemProperties();
+
+        return configuration;
+    }
+
+    static Configuration withDefaultValues() {
+        final Configuration configuration = new Configuration();
+
+        configuration.mode = RunMode.SELECTING;
+        configuration.report = Report.fromDefaultValues();
+        configuration.scm = Scm.fromDefaultValues();
+        configuration.debug = false;
+        configuration.disable = false;
+
+        return configuration;
+    }
+
+    public static Configuration load() {
+        return load(Paths.get("").toAbsolutePath().toFile());
+    }
+
+    public static Configuration load(File projectDir) {
+        final File[] files =
+            projectDir.listFiles((dir, name) -> name.equals(SMART_TESTING_YML) || name.equals(SMART_TESTING_YAML));
+
+        Map<String, Object> yamlConfiguration = new LinkedHashMap<>();
+
+        if (files != null) {
+            if (files.length == 0) {
```

Comment:
these two `if`s should be merged, to handle (log) also the case when `files` is `null`